ipn/ipnlocal/peerapi.go

```diff
@@ -341,11 +341,71 @@ func (s *peerAPIServer) DeleteFile(baseName string) error {
 // accidentally logging actual filenames anywhere.
 const redacted = "redacted"
 
+type redactedErr struct {
+	msg   string
+	inner error
+}
+
+func (re *redactedErr) Error() string {
+	return re.msg
+}
+
+func (re *redactedErr) Unwrap() error {
+	return re.inner
+}
+
 func redactErr(err error) error {
-	if pe, ok := err.(*os.PathError); ok {
+	var redactStrings []string
+
+	var pe *os.PathError
+	if errors.As(err, &pe) {
+		// If this is the root error, then we can redact it directly.
+		if err == pe {
+			pe.Path = redacted
```

Comment:
Would it be worth appending some hash of the original value so that we can distinguish between them?

Reply:
Yeah, good idea; perhaps something like `hash/maphash.Hash.Sum64()` with a constant per-process `Seed`? Thoughts?

Reply:
I was thinking something simpler like just `adler32.Checksum` so that it's deterministic across nodes.